zeeguu_api/api/user_languages.py

```diff
@@ -0,0 +1,155 @@
+import flask
+import zeeguu
+from flask import request
+from zeeguu.model.language import Language
+from zeeguu.model.user_language import UserLanguage
+
+
+from .utils.route_wrappers import cross_domain, with_session
+from .utils.json_result import json_result
+from . import api
+
+session = zeeguu.db.session
+
+USER_LANGUAGES = "user_languages"
+MODIFY_USER_LANGUAGE = "user_languages/modify"
+DELETE_USER_LANGUAGE = "user_languages/delete"
+INTERESTING_LANGUAGES = "user_languages/interesting"
+READING_LANGUAGES = "user_languages/reading"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{MODIFY_USER_LANGUAGE}", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def add_user_language():
+    """
+    This endpoint is for modifying a user language.
+    It tries to find the user_language, and otherwise create it.
+    It then sets all the parameters given.
+
+    :return: "OK" in case of success
+    """
+    language_id = int(request.form.get('language_id', ''))
+    try:
+        language_reading = int(request.form.get('language_reading', ''))
+    except:
+        language_reading = None
+    try:
+        language_exercises = int(request.form.get('language_exercises', ''))
+    except:
+        language_exercises = None
+    try:
+        language_level = int(request.form.get('language_level', ''))
+    except:
+        language_level = None
+
+    language_object = Language.find_by_id(language_id)
+    user_language = UserLanguage.find_or_create(session, flask.g.user, language_object)
+    if language_reading is not None:
+        user_language.reading_news = language_reading
+    if language_exercises is not None:
+        user_language.doing_exercises = language_exercises
+    if language_level is not None:
+        user_language.declared_level = language_level
+    session.add(user_language)
+    session.commit()
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{DELETE_USER_LANGUAGE}/<language_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def delete_user_language(language_id):
+    """
+    A user can delete a language with a given ID.
+    With this endpoint the full user language with all the data is deleted.
+    At the moment not used but might be useful.
+
+    :return: OK / ERROR
```

Comment:
better comment for return is `"OK" in case of success`

Reply:
Changed